adapters/adapter.go

```diff
@@ -37,7 +37,7 @@ func (wa MinConfsWrappedAdapter) MinConfs() uint64 {
 }
 
 // For determines the adapter type to use for a given task.
-func For(task models.TaskSpec, store *store.Store) (AdapterWithMinConfs, error) {
+func For(task *models.TaskSpec, store *store.Store) (AdapterWithMinConfs, error) {
```

Comment:
Hey there @j16r, any specific reason why we're changing `TaskSpec` and `Initiator` to be passed by reference? This makes it more likely to be a mutable type and all the method receivers, with the exeption of `UnmarshalJSON`, take a value to encourage immutability. We're doing this for a variety of reasons (prevent sharing state, especially if it's passed among multiple go-routines, which it is). We can chat about whether or not this is the right direction offline if you want.

If you do want to modify one, you can have it return a new copy like so:

https://github.com/smartcontractkit/chainlink/blob/9e5a66596c2b642bf9d138505cd2984d46cb8329/services/job_runner.go#L112

```golang
jr = jr.ApplyResult(latestRun.Result)
...
// ApplyResult updates the JobRun's Result and Status
func (jr JobRun) ApplyResult(result RunResult) JobRun {
	jr.Result = result
	jr.Status = result.Status
	if jr.Status.Completed() {
		jr.CompletedAt = null.Time{Time: time.Now(), Valid: true}
	}
	return jr
}
```

Reply:
This was required by one of the JSON libraries I was using, but it's not currently required by the one I've finalized this with so I will look into removing.